lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -74,21 +74,21 @@ module Deserialization
         #
         def parse!(document, options = {})
           parse(document, options) do |invalid_payload, reason|
-            fail InvalidDocument, "Invalid payload (#{reason}): #{invalid_payload}"
+            fail JSONAPI::Parser::InvalidDocument, "Invalid payload (#{reason}): #{invalid_payload}"
           end
         end
 
         # Same as parse!, but returns an empty hash instead of raising InvalidDocument
         # on invalid payloads.
         def parse(document, options = {})
-          document = document.dup.permit!.to_h if document.is_a?(ActionController::Parameters)
-
-          validate_payload(document) do |invalid_document, reason|
-            yield invalid_document, reason if block_given?
-            return {}
-          end
+          document = JSONAPI.parse(document, options)
+          document = document.to_hash
```

Comment:
I think the validator could be interesting as long as it was quick. In my project, I had to disable validation, cause I am doing embedded records in relationships.

I was looking at jsonapi-deserializable the other day. It errored, so I didn't add it to this PR.